### PR TITLE
Set uses_from_macos_elements in all cases

### PR DESCRIPTION
**Pull request by Bo98**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

Hopefully fixes CI. Kinda rushed this so not tested.

TODO: adjust any `uses_from_macos_elements` checks to also consider the case where the element is a hash (e.g. `"dep" => :build`).

### CODE REVIEW

**Comment by Bo98** [comment]
> TODO: adjust any `uses_from_macos_elements` checks to also consider the case where the element is a hash (e.g. `"dep" => :build`).

This is also where it starts to overlap a bit with how DependencyCollector works.

File: Library/Homebrew/language/perl.rb [hunk 1]
```diff
@@ -11,10 +11,12 @@ module Shebang
...
 
       def detected_perl_shebang(formula = self)
-        perl_path = if formula.uses_from_macos_elements&.include? "perl"
-          "/usr/bin/perl#{MacOS.preferred_perl_version}"
-        elsif formula.deps.map(&:name).include? "perl"
+        perl_path = if formula.deps.map(&:name).include? "perl"
           Formula["perl"].opt_bin/"perl"
+        elsif formula.uses_from_macos_elements
+                     .flat_map { |e| e.is_a?(Hash) ? e.keys : e }
+                     .include? "perl"
```
Comment (Bo98) [inline Library/Homebrew/language/perl.rb:18]: Thoughts on this for the `formula_auditor` check too?

Comment (MikeMcQuaid) [reply Library/Homebrew/language/perl.rb]: @Bo98 Hah, started on this before I saw the comment. Pulled into a method.

**Comment by MikeMcQuaid** [comment]
Thanks @Bo98!

**Pull request merged**